### DISCUSSION TITLE
Eval [(expr)] prevented in JSONPath expression.

### DISCUSSION THREAD

**Discussion by yogeshboddu** [grafana/grafana-json-datasource#341] [Q&A]
Getting the below error while querying the data
Query error
Eval [(expr)] prevented in JSONPath expression.

<img width="1068" alt="image" src="https://user-images.githubusercontent.com/95027376/230081646-ea594da5-c015-4941-a495-03c7ffadede1.png">

**Comment by pacallan** [comment]
> Getting the below error while querying the data Query error Eval [(expr)] prevented in JSONPath expression.
> 
> <img alt="image" width="1068" src="https://user-images.githubusercontent.com/95027376/230081646-ea594da5-c015-4941-a495-03c7ffadede1.png">

We are seeing this too. It only started very recently, and we did not change anything at our end.

Edit:
We are using Grafana Cloud  which is currently Grafana v9.5.0 (769132c5eb).
The plug-in was at v1.3.4 when the problem started, but I have since upgraded to v1.3.5 (the latest) but the issue still occurs.
![Template errors on Grafana](https://user-images.githubusercontent.com/8492783/230335681-56bccb00-68ba-4b70-a258-6b019df4a86c.JPG)

**Comment by anthonypruiz** [comment]
Same Here! broke dashboards, no [?(expr)] working for JSON API

**Comment by Wepstern** [comment]
We got the same issue. No changes has been made on Grafana and JSON API side.

**Comment by NickSmet** [comment]
I've rolled back to version v1.3.2. 
Works fine.

**Comment by pacallan** [comment]
It seems to have been flagged as a bug here:
https://github.com/grafana/grafana-json-datasource/issues/343

v1.3.3 and below are fine. Bug was introduced at v1.3.4.

Not sure what is being done about it.

**Comment by danjuggles** [comment]
Any got an update on this, (or a workaround?) I am still seeing this exact issue. It's really annoying!

**Reply by pacallan** [reply]
Have a look here for workarounds etc. https://github.com/grafana/grafana-json-datasource/issues/343

We have moved to JSONata for evaluation of expressions, still using the JSON API Plugin for most things.
